File: ARC120/A.py
# -*- coding: utf-8 -*-
 
 
def a_hoge(A):
    size = len(A)
    for i in range(size):
        A[i]+=max(A)
    print((A))
    
    
    
 
n = int(input())
 
A = list(map(int, input().split()))
k=1
 
mae = 0
for i in range(n):
     
    A[i]+=max(A[:k])
    
    hoge = sum(A[:k])
    
    print(hoge+mae)
    
    mae = hoge
    k+=1
 
 

File: ARC120/A.cc
#include "iostream"
#include <algorithm>
#include <numeric>

int main(){
  long long int n, max, array[200001] = {0}, i, mae = 0, j, sum;

  std::cin >> n;

  for(i=0; i<n; ++i){
    std::cin >> array[i];

    array[i] += *std::max_element(array, array+200001);

    sum = std::accumulate(array, array+200001, 0);
    std::cout << mae+ sum << '\n';
    mae = sum;


  }

  return 0;
}
